### PR TITLE
Hijacking a shuttle diverts it to the syndicate outpost

### PR DESCRIPTION
**Pull request by tigercat2000**
tgstation/tgstation#22184

 - Hijacking the shuttle will make it dock to a new dock on the
 syndicate outpost.
 - Fixed shuttle_master not calling mobile/dockRoundstart().
 - Only Traitors, Changelings, Vampires, Nuclear Operatives, and
 Syndicate Deathsquads can now hijack the shuttle.

### CODE REVIEW

**Comment by Spacemanspark** [comment]
Not bad.

**Comment by tigercat2000** [comment]
Divers it? Yeah, the shuttle is a grammatically incorrect submarine.

**Comment by AndriiYukhymchak** [comment]
Thanks for implementing it. I suggested it long time ago. Now people with hijack should just insert their am at into console, and not kill everyone

**Comment by FlattestGuitar** [comment]
You still need to get rid of every nerd onboard the shuttle for it to dock in the Syndie base.

Not easy.

**Comment by Spacemanspark** [comment]
With our super duper cheese method it is.

File: code/modules/shuttle/emergency.dm [hunk 1]
```diff
@@ -166,12 +166,36 @@
...
+		if(player.mind)
+			if(player.stat != DEAD)
+				if(issilicon(player)) //Borgs are technically dead anyways
+					continue
+				if(isanimal(player)) //Poly does not own the shuttle
+					continue
+
+				var/special_role = player.mind.special_role
+				if(special_role)
+					if(special_role == SPECIAL_ROLE_TRAITOR) // traitors can hijack the shuttle
```
Comment (Fox-McCloud) [inline code/modules/shuttle/emergency.dm]: just to be sure, but mindslaves also fall under this category, correct?

Comment (tigercat2000) [inline code/modules/shuttle/emergency.dm]: [Yep!](https://github.com/ParadiseSS13/Paradise/blob/master/code/game/objects/items/weapons/implants/implant_traitor.dm#L57)

Comment (Fox-McCloud) [inline code/modules/shuttle/emergency.dm]: Thankssss

**Comment by tigercat2000** [comment]
@Crazylemon64 Updated.

File: code/modules/shuttle/emergency.dm [hunk 2]
```diff
@@ -166,12 +166,36 @@
...
-/obj/docking_port/mobile/emergency/findTransitDock()
-	. = shuttle_master.getDock("emergency_transit")
-	if(.)	return .
-	return ..()
-*/
+/obj/docking_port/mobile/emergency/proc/is_hijacked()
+	for(var/mob/living/player in player_list)
+		if(player.mind)
+			if(player.stat != DEAD)
+				if(issilicon(player)) //Borgs are technically dead anyways
```
Comment (Crazylemon64) [inline code/modules/shuttle/emergency.dm]: Wait, does this mean that if the shuttle is piloted entirely by cyborgs, that the shuttle will head to the syndicate base :smirk:

Comment (tigercat2000) [inline code/modules/shuttle/emergency.dm]: feature:tm:

**Comment by Fox-McCloud** [comment]
Merge conflictssss

**Comment by tigercat2000** [comment]
👌

**Comment by Nikitaw99** [comment]
👍 Sounds fun. 👍

**Pull request merged**